extensions/mssql/src/iconProvider.ts

```diff
@@ -7,14 +7,18 @@ import * as azdata from 'azdata';
 import * as constants from './constants';
 
 const cloudIcon = 'mssql:cloud';
+const miIcon = "mssql:miIcon";
 
 export class MssqlIconProvider implements azdata.IconProvider {
 	public readonly providerId: string = constants.sqlProviderName;
 	public handle: number;
 	getConnectionIconId(connection: azdata.IConnectionProfile, serverInfo: azdata.ServerInfo): Thenable<string> {
 		let iconName: string = undefined;
 		if (connection.providerName === 'MSSQL') {
-			if (serverInfo.isCloud) {
+			if (serverInfo.engineEditionId === 8) {
```

Comment:
Is it safe to assume this edition is always only MI?

Reply:
Yes, it's safe. Engine edition 8 means its Managed Instance. 
https://learn.microsoft.com/en-us/sql/t-sql/functions/serverproperty-transact-sql?view=sql-server-ver16